### PR TITLE
Generified AttestationStatementValidators

### PR DESCRIPTION
**Pull request by kvandermast**
Moved the "supports" method to a abstract class

### CODE REVIEW

**Comment by ynojima** [comment]
Interesting refactoring.

File: webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/AbstractStatementValidator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public abstract class AbstractStatementValidator<T extends AttestationStatement> implements AttestationStatementValidator {
+    @Override
+    public boolean supports(RegistrationObject registrationObject) {
+        AttestationStatement attestationStatement = registrationObject.getAttestationObject().getAttestationStatement();
+
+        ParameterizedType parameterizedType = (ParameterizedType) getClass().getGenericSuperclass();
+        Type actualTypeArgument = parameterizedType.getActualTypeArguments()[0];
+
+        if (actualTypeArgument instanceof Class) {
+            Class<?> aClass = (Class<?>) actualTypeArgument;
```
Comment (ynojima) [inline webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/AbstractStatementValidator.java]: `aClass` is better to be initialized in the constructor. No need to be calculated every time `supports` method called.

**Comment by kvandermast** [comment]
Thanks, the idea was - initially - to use the generic when retrieving the attestation statement, so that the abstract class would have this

```
public <T> getAttestationStatement() {
   // ... 
}
```

I didn't manage to get to that part yet, but this seems like a potential refactoring candidate to reduce some implementation.

Putting it in the constructor seems a good approach indeed.

File: webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/AbstractStatementValidator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+            return aClass.isAssignableFrom(attestationStatement.getClass());
+        }
+
+        return false;
```
Comment (ynojima) [inline webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/AbstractStatementValidator.java]: When does the execution reach here? I suppose it is better to throw `IllegalStateException` with message describing the situation.

Comment (kvandermast) [reply webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/AbstractStatementValidator.java]: In theory, never. 
- T must be of type AttestationStatement
- If the resolved type is not a class (covered by the above)

So an IllegalStateException might be better indeed.

**Pull request merged**